### ISSUE TITLE
Replace Metrics.Net with AppMetrics

### ISSUE THREAD

**Issue by edwardmeng** [FoundatioFx/Foundatio#114]
The library Metrics.Net is only available for netfx and only can be deployed on windows servers. But the Foundatio currently supports netstandard 2.0, the Metrics.Net should be replaced with AppMetrics which supports netstandard, so that we can deployed on linux or unix.

**Comment by ejsmith** [comment]
@edwardmeng I agree. I have been wanting to make some changes to the Foundatio metrics contract to allow adding tags/dimensions to metrics and I also wanted to create an AppMetrics implementation as well.

**Comment by niemyjski** [comment]
They are working on a metrics.net net standard implementation and it was user contribed (on a side note). I also agree with the both of you but against removing metrics.net :)

**Comment by ejsmith** [comment]
Wouldn't remove it. Would just add another implementation.

**Comment by edwardmeng** [comment]
@ejsmith I am wondering how to support tags/dimensions for statsd implementation. As I know the statsd does not support it be default.

**Comment by ejsmith** [comment]
Sorry for the late reply @edwardmeng. Yeah, there are some implementations that would not be able to to support tagging/dimensions. So those would just be ignored. I am envisioning that we just add some methods to let you create scopes that contain the dimensions on them and they just return the normal IMetricsClient back so none of the core contract would change. It would just be a matter of adding methods like this:

```csharp
public interface IMetricsClient : IDisposable {
   IMetricsClient BeginScope(IDictionary<string, object> data);
   void Counter(string name, int value = 1);
   void Gauge(string name, double value);
   void Timer(string name, int milliseconds);
}
```

Then we would have extension methods with a fluent builder that let you build up the scope dictionary using a fluent lambda syntax which would be used something like this:

```csharp
using (var scoped = client.StartScope(s => s.Tag("mytag").Dimension("host", "myhost1"))) {
  scoped.Counter("mycounter");
}
```

Tags would just be added to the dictionary without any values.

**Comment by niemyjski** [comment]
Thanks for this issue, we've removed Metrics as we use OTEL.
